diff --git a/module/commerce_checkout/fz152_commerce_checkout.info b/module/commerce_checkout/fz152_commerce_checkout.info
new file mode 100644
index 0000000..1de4de9
--- /dev/null
+++ b/module/commerce_checkout/fz152_commerce_checkout.info
@@ -0,0 +1,8 @@
+name = FZ152 — Commerce Checkout
+description = Integration with Commerce Checkout.
+core = 7.x
+
+package = FZ152
+
+dependencies[] = fz152
+dependencies[] = commerce_checkout
diff --git a/module/commerce_checkout/fz152_commerce_checkout.module b/module/commerce_checkout/fz152_commerce_checkout.module
new file mode 100644
index 0000000..9c6da13
--- /dev/null
+++ b/module/commerce_checkout/fz152_commerce_checkout.module
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Implements hook_commerce_checkout_pane_info().
+ */
+function fz152_commerce_checkout_commerce_checkout_pane_info() {
+  $checkout_panes = array();
+
+  $checkout_panes['fz152_commerce_checkout_pane'] = array(
+    'name' => t('FZ152 pane'),
+    'base' => 'fz152_commerce_checkout_pane',
+    'page' => 'checkout',
+    'locked' => FALSE,
+    'review' => FALSE,
+    'weight' => 99,
+  );
+
+  return $checkout_panes;
+}
+
+/**
+ * Checkout pane callback: returns a pane with checkbox inside.
+ */
+function fz152_commerce_checkout_pane_checkout_form($form, &$form_state, $checkout_pane, $order) {
+  $pane_form = array();
+
+  // Otherwise add any enabled checkout panes that are visible on review.
+  $pane_form['fz152_agreement'] = array(
+    '#type' => 'checkbox',
+    '#required' => TRUE,
+    '#title' => variable_get_value('fz152_checkbox_title'),
+    // HTML5 support.
+    /* '#attributes' => array(
+      'required' => 'required',
+    ),*/
+  );
+
+  return $pane_form;
+}
